### PR TITLE
Add test for indexdb composable

### LINKED ISSUE #1729: Tests für IndexDB-Composable erstellen
Für das composable `indexDB` wurden im Rahmen des Prototypen #1407 keine Unit-Tests erstellt. Dies soll jetzt nachgeholt werden.

# Akzeptanzkriterien
- [ ] es wurden unit-Tests zu dem Composable erstellt

### PR DESCRIPTION
**Pull request by ThoGro**
# Beschreibung:

Test für IndexDB Composable hinzugefügt.

### Frontend
- [x] Unit-Tests gepflegt

# Referenzen[^1]:

Closes #1729

> [^1]: _Nicht zutreffende Referenzen vor dem Speichern entfernen_

[naming-conventions-link]: https://it-at-m.github.io/Wahllokalsystem/technik/naming_conventions/
[fachliche-beschreibung-link]: https://it-at-m.github.io/Wahllokalsystem/about/#fachliche-anforderungen
[ui-ux-adrs-link]: https://it-at-m.github.io/Wahllokalsystem/technik/adr/ui/

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive unit tests for the local data storage layer, covering retrieval of pending changes, item lookups, backend configuration, and data saving.
  * Strengthens error handling and validates behavior across edge cases, improving stability and reliability in offline workflows.
  * Increases test coverage to help prevent regressions and ensure consistent user experience.

### CODE REVIEW

File: wls-gui-wahllokalsystem/tests/composables/indexDB/indexDB.spec.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      vi.spyOn(localforage, "getItem").mockRejectedValue(error);
+
+      const result = await unitUnderTest.getItemFromIDB(key);
+
+      expect(localforage.getItem).toHaveBeenCalledWith(key);
+      expect(consoleMock).toHaveBeenCalledWith(
+        "useIndexDB: Fehler beim Laden aus IDB:",
+        error
+      );
+      expect(result).toEqual(null);
```
[the remark this answers is not in the document]
Comment (ThoGro) [reply wls-gui-wahllokalsystem/tests/composables/indexDB/indexDB.spec.ts:104]: [0128557](https://github.com/it-at-m/Wahllokalsystem/pull/1932/commits/0128557ff3f9c485d1280077afc00e6f1528b1b7)

**Review by vjohnslhm — DISMISSED** [review]

**Review by vjohnslhm — APPROVED** [review]
